### PR TITLE
Move all "old" SQL operators to common.sql providers

### PR DESCRIPTION
**Pull request by potiuk**
Previously, in #24836 we moved Hooks and added some new operators to the
common.sql package. Now we are salso moving the operators
and sensors to common.sql.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/operators/check_operator.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 import warnings
 
-from airflow.operators.sql import (
+from airflow.providers.common.sql.operators.sql import (
```
Comment (josh-fell) [inline airflow/operators/check_operator.py:23]: Should the deprecation warnings in this file and others also reflect the new import path to use?

Comment (potiuk) [reply airflow/operators/check_operator.py:23]: Very much so @josh-fell :). Fixed.

File: airflow/providers/apache/druid/operators/druid_check.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 # under the License.
 import warnings
 
-from airflow.operators.sql import SQLCheckOperator
+from airflow.providers.common.sql.operators.sql import SQLCheckOperator
```
Comment (josh-fell) [inline airflow/providers/apache/druid/operators/druid_check.py:20]: One more deprecation message is hiding for `DruidCheckOperator`.

File: airflow/operators/druid_check_operator.py [hunk 3]
```diff
@@ -22,7 +22,7 @@
 from airflow.providers.apache.druid.operators.druid_check import DruidCheckOperator  # noqa
 
 warnings.warn(
-    "This module is deprecated. Please use `airflow.operators.sql.SQLCheckOperator`.",
+    "This module is deprecated. Please use `airflow.providers.apache.druid.operators.druid_check` module.",
```
Comment (josh-fell) [inline airflow/operators/druid_check_operator.py]:
```suggestion
    "This module is deprecated. Please use `airflow.providers.common.operators.sql` module.",
```
Looks like the `DruidCheckOperator` and the module is also deprecated and points to the `SQLCheckOperator`. Deprecation inception.

Comment (potiuk) [reply airflow/operators/druid_check_operator.py]: Ah. That's correct we did not need DruidCheck - plain SQLcheck was enough passing Druid Hook.  So all is good after my corrections.

**Comment by potiuk** [comment]
Hopefully Green now.

File: airflow/providers/common/sql/CHANGELOG.rst [hunk 4]
```diff
@@ -24,6 +24,11 @@
 Changelog
 ---------
 
+1.1.0
```
Comment (eladkal) [inline airflow/providers/common/sql/CHANGELOG.rst:27]: Did you mean to add it without the log entry of the change?

File: airflow/operators/sql.py [hunk 5]
```diff
@@ -15,543 +15,23 @@
...
-        """
-        Get the database hook for the connection.
-
-        :return: the database hook object.
-        :rtype: DbApiHook
-        """
-        return self._hook
-
-
-class SQLCheckOperator(BaseSQLOperator):
```
Comment (eladkal) [inline airflow/operators/sql.py:91]: ~Removing the operators is a breaking change.... We need to deprecate them~
Ah you did that :)
I missed it

**Review by eladkal — APPROVED** [review]

File: airflow/providers/common/sql/CHANGELOG.rst (hunk 4 above)
Comment (potiuk) [reply airflow/providers/common/sql/CHANGELOG.rst:27]: Yes. I need it in order to get the condition fullfilled `>= 1.1.0.*` when building prod images. I will add more description when I prepare the final release.

**Pull request merged**